src/utilcode/prettyprintsig.cpp

```diff
@@ -688,52 +692,21 @@ static HRESULT PrettyPrintTypeA(
         
     case ELEMENT_TYPE_CMOD_REQD:
         str = "required_modifier ";
-        goto DO_CLASS;
+        IfFailGo(appendStrA(out, str));
+        IfFailGo(PrettyPrintClass(typePtr, out, pIMDI));                  
+        IfFailGo(appendStrA(out, " "));
+        IfFailGo(PrettyPrintTypeA(typePtr, (typeEnd - typePtr), out, pIMDI));
+        break;
         
     case ELEMENT_TYPE_CMOD_OPT:
```

Comment:
Could you please do the same thing for `ELEMENT_TYPE_CMOD_REQD` as well?

Reply:
Sorry, I'm not following. Did you mean calling `PrettyPrintClass` also for `ELEMENT_TYPE_CMOD_OPT`? Or otherwise what should I change for `ELEMENT_TYPE_CMOD_REQD`?

Reply:
> Did you mean calling PrettyPrintClass also for `ELEMENT_TYPE_CMOD_OPT`? 

Yes, that's what I meant. The change for `ELEMENT_TYPE_CMOD_REQD` case is good. I meant to ask to make the same change for `ELEMENT_TYPE_CMOD_OPT`` case.

Reply:
Got it. Is there a C# method signature that would trigger the `ELEMENT_TYPE_CMOD_OPT` case so I can test it?

Reply:
I do not think there is a case today where C# uses modopts. 

You can test it by modifying the: take your current test for modreq, `ildasm test.exe`, change modreq to modopt in test.il, `ilasm test.exe`.

Reply:
Good idea, optional_modifier works as well now (outputs `optional_modifier System.Runtime.InteropServices.InAttribute int64&  (int32)`).